kernel/api/src/main/java/org/sakaiproject/util/api/ValueEncryptionService.java

```diff
@@ -0,0 +1,21 @@
+package org.sakaiproject.util.api;
+
+public interface ValueEncryptionService {
```

Comment:
Mmmmm... but now the providers implementations are inside kernel...

Then, are you suggesting to move the providers implementations inside providers and let in kernel... only the ChainedDetailProvider?

Are you right with that?